### ISSUE TITLE
broken with ghc-8.10

### ISSUE THREAD

**Issue by mimi1vx** [arowM/heterocephalus#26]
```
[6 of 6] Compiling Text.Heterocephalus ( src/Text/Heterocephalus.hs, dist/build/Text/Heterocephalus.o, dist/build/Text/Heterocephalus.dyn_o )

src/Text/Heterocephalus.hs:525:22: error:
    • Couldn't match expected type ‘Maybe Exp’ with actual type ‘Exp’
    • In the expression: d'
      In the first argument of ‘TupE’, namely ‘[d', docs']’
      In the second argument of ‘($)’, namely ‘TupE [d', docs']’
    |
525 |       return $ TupE [d', docs']
    |                      ^^

src/Text/Heterocephalus.hs:525:26: error:
    • Couldn't match expected type ‘Maybe Exp’ with actual type ‘Exp’
    • In the expression: docs'
      In the first argument of ‘TupE’, namely ‘[d', docs']’
      In the second argument of ‘($)’, namely ‘TupE [d', docs']’
    |
525 |       return $ TupE [d', docs']
    |                          ^^^^^
CallStack (from HasCallStack):
  die', called at ./Distribution/Client/ProjectOrchestration.hs:1041:55 in main:Distribution.Client.ProjectOrchestration
cabal: Failed to build
```

**Comment by mimi1vx** [comment]
@arowM something for inspiration :+1: 

https://github.com/ekmett/free/pull/195/commits/1d594d9d11ae42a71e2c1c3dc3a37acbb608c33c <-- fix for same problem with compactibility layer in ekmett's `free` package

**Comment by arowM** [comment]
thanks

**Comment by arowM** [comment]
If anyone can fix this for me, I would welcome it.
Or I can do it when I have time.

**Comment by mimi1vx** [comment]
I'll try fix it on weekend ..., termonad is for me best terminal app :D and this is ony blocker with newer ghc versions.
@cdepillabout  .. we started packaging GHC-9.0.x to openSUSE and nixOS.

**Comment by arowM** [comment]
Really? Thank you so much！ 💐

**Comment by mimi1vx** [comment]
fix in #27
